rpc: TestRemoteOffsetUnhealthy failed

rpc.TestRemoteOffsetUnhealthy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13243328?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13243328?buildTab=artifacts#/) on master @ [7a7bbffa8959c9004991602b9d124de365aa8a17](https://github.com/cockroachdb/cockroach/commits/7a7bbffa8959c9004991602b9d124de365aa8a17):


```
=== RUN   TestRemoteOffsetUnhealthy
I231220 21:19:37.955065 402 2@rpc/peer.go:527  [rnode=2,raddr=127.0.0.1:44837,class=default,rpc] 19  connection is now healthy
I231220 21:19:37.969569 408 2@rpc/peer.go:527  [rnode=3,raddr=127.0.0.1:41535,class=default,rpc] 20  connection is now healthy
I231220 21:19:37.975695 97 2@rpc/peer.go:527  [rnode=4,raddr=127.0.0.1:34655,class=default,rpc] 21  connection is now healthy
I231220 21:19:37.982762 411 2@rpc/peer.go:527  [rnode=1,raddr=127.0.0.1:38141,class=default,rpc] 22  connection is now healthy
I231220 21:19:37.996518 440 2@rpc/peer.go:527  [rnode=3,raddr=127.0.0.1:41535,class=default,rpc] 23  connection is now healthy
I231220 21:19:38.003363 468 2@rpc/peer.go:527  [rnode=4,raddr=127.0.0.1:34655,class=default,rpc] 24  connection is now healthy
I231220 21:19:38.009764 499 2@rpc/peer.go:527  [rnode=1,raddr=127.0.0.1:38141,class=default,rpc] 25  connection is now healthy
I231220 21:19:38.015776 514 2@rpc/peer.go:527  [rnode=2,raddr=127.0.0.1:44837,class=default,rpc] 26  connection is now healthy
E231220 21:19:38.136548 521 2@rpc/peer.go:601  [rnode=4,raddr=127.0.0.1:34655,class=default,rpc] 27  failed connection attempt (last connected 0s ago): operation "conn heartbeat" timed out after 120ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
    context_test.go:1172: initial connection heartbeat failed: operation "conn heartbeat" timed out after 120ms (given timeout 100ms): grpc: context deadline exceeded [code 4/DeadlineExceeded]
--- FAIL: TestRemoteOffsetUnhealthy (0.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRemoteOffsetUnhealthy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34836

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -1163,9 +1163,16 @@ func TestRemoteOffsetUnhealthy(t *testing.T) {
 			if i == j {
 				continue
 			}
-			if _, err := clientNodeContext.ctx.GRPCDialNode(serverNodeContext.ctx.AdvertiseAddr, serverNodeContext.ctx.NodeID.Get(), DefaultClass).Connect(ctx); err != nil {
-				t.Fatal(err)
-			}
+			testutils.SucceedsSoon(t, func() error {
+				if _, err := clientNodeContext.ctx.GRPCDialNode(
+					serverNodeContext.ctx.AdvertiseAddr,
+					serverNodeContext.ctx.NodeID.Get(),
+					DefaultClass,
+				).Connect(ctx); err != nil {
+					return err
+				}
+				return nil
+			})
 		}
 	}
 

